### PR TITLE
Added support for SendGrid and standard SMTP

### PR DESCRIPTION
**Pull request by fegu**
including docs

### CODE REVIEW

File: IHP/Mail.hs [hunk 1]
```diff
@@ -19,10 +19,14 @@ import IHP.FrameworkConfig
 
 import           Network.Mail.Mime
 import qualified Network.Mail.Mime.SES                as Mailer
+import qualified Network.Mail.SMTP                    as SMTP
 import qualified Network.HTTP.Client
 import qualified Network.HTTP.Client.TLS
 import Text.Blaze.Html5 (Html)
 import qualified Text.Blaze.Html.Renderer.Text as Blaze
+import qualified Data.ByteString.Lazy.Char8 as B8
+import qualified Data.Text as T
```
Comment (mpscholten) [inline IHP/Mail.hs:28]:
```suggestion
import qualified Data.Text as Text
```

File: IHP/Mail.hs [hunk 2]
```diff
@@ -19,10 +19,14 @@ import IHP.FrameworkConfig
 
 import           Network.Mail.Mime
 import qualified Network.Mail.Mime.SES                as Mailer
+import qualified Network.Mail.SMTP                    as SMTP
 import qualified Network.HTTP.Client
 import qualified Network.HTTP.Client.TLS
 import Text.Blaze.Html5 (Html)
 import qualified Text.Blaze.Html.Renderer.Text as Blaze
+import qualified Data.ByteString.Lazy.Char8 as B8
```
Comment (mpscholten) [inline IHP/Mail.hs:27]:
```suggestion
import qualified Data.ByteString.Lazy.Char8 as LazyByteString
```

Comment (mpscholten) [reply IHP/Mail.hs:27]: The IHP code mostly follows these conventions when using qualified imports. To make the code feel consistent I suggest changing this :)

File: IHP/Mail/Types.hs [hunk 3]
```diff
@@ -9,17 +9,22 @@ module IHP.Mail.Types
...
         --  | E.g. @"us-east-1"@ or @"eu-west-1"@
         , region :: Text }
-    -- | Uses the local Sendmail binary for sending emails
+    -- | Uses the local Sendmail binary for sending emails. Avoid this with IHP Cloud
     | Sendmail
+    -- | Uses SendGrid for sending emails
+    | SendGrid { apiKey :: Text
+               , category :: Maybe Text }
+    -- | Uses a generic SMTP for sending emails
+    | GenericSMTP { host :: String
```
Comment (mpscholten) [inline IHP/Mail/Types.hs:27]: Can't we just name this `SMTP`?

```suggestion
    | SMTP { host :: String
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -99,6 +99,8 @@ The documentation reads a bit like a tutorial, but should still be kept somewhat
 
 - Please add Haddock-comments to new methods intended to be used by directly when developing using IHP.
 
+- Please consider carefully before adding new packages as requirements to IHP itself. Make sure the packages are actively maintained.
```
Comment (mpscholten) [inline CONTRIBUTING.md:102]: 👍

**Comment by mpscholten** [comment]
Really good addition :) Thanks!

**Comment by fegu** [comment]
Merry christmas!

**Review by mpscholten — APPROVED** [review]
Perfect thanks :) have a happy christmas!

**Comment by sebastian-palma** [comment]
There's an error when trying to compile IHP/Mail.hs (for development):

```haskell
IHP/IHP/Mail.hs:22:1: error:
    Could not find module ‘Network.Mail.SMTP’
    Perhaps you meant Network.Mail.Mime (from mime-mail-0.5.0)
    Use -v (or `:set -v` in ghci) to see a list of the files searched for.
   |
22 | import qualified Network.Mail.SMTP                    as SMTP
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Am I doing something wrong? Skipping something?

**Comment by fegu** [comment]
Until the next beta is released you need to add package smtp-mail to your default.nix.

**Comment by mpscholten** [comment]
Alternative you can also do this in your default.nix:
```nix
let
    ihp = ./IHP;
    haskellEnv = ...
```

**Pull request merged**